server/routerlicious/packages/services-core/src/storageNameRetriever.ts

```diff
@@ -0,0 +1,8 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export interface IStorageNameRetriever {
+	get(tenantId: string, documentId: string): Promise<string>;
+}
```

Comment:
`IStorageNameProvider` from previous version seems sufficient? what is the semantics difference between `IStorageNameProvider` and `IStorageNameAllocator`?

Reply:
Previous the `IStorageNameProvider` contains both `read` and `write`, and now I split into two for single responsibility concern. Therefore used `IStorageNameAllocator` to indicate this is for create/write only

Reply:
Any benefits to separate read and write into 2 classes?

Reply:
It makes sense to have an interface for reading / obtaining storage name since many components need this function and it's good to have a unified API for it, but it'd make less sense to have an interface for generating the storage name, since it'd be only used by the component that creates a tenant or a document.

Reply:
After checking out the PR for FRS, found out you have implemented more than what I expected :). `IStorageNameAllocator` makes more sense now since it abstracts the storage name allocation strategies in R11S and FRS (rather than among components). It also makes sense to separate `IStorageNameAllocator` and `IStorageNameRetriever` since they aren't used by the same components (Alfred would use the former while other components use the latter).